### DISCUSSION TITLE
Writing a custom Connector (General Help)

### DISCUSSION THREAD

**Discussion by mrjavascript** [nextgenhealthcare/connect#4995] [Q&A] [answered]
I'm writing a custom connector (source and destination).  I have the project building a zip file which I import via extension manager in the admin tool.  I modified the starter here:

https://github.com/kpalang/mirth-plugin-guide

I got the build including the server, client and shared jar in the zip.  The source.xml and destination.xml are also included and I'm able to import it without errors.  When I go to configure a new channel and set the new connector, something fails:

![image](https://user-images.githubusercontent.com/2151084/152422517-a30ff0a1-abf4-450a-a957-b312d870faa4.png)

Is there a setting I can apply to expose more logging when this screen fails?  My source.xml has the following content:

    <description>This connector allows Mirth Connect to send messages to an Amazon AWS SQS queue.</description>
    <clientClassName>com.foo.client.connectors.sqs.SqsWriter</clientClassName>
    <serverClassName>com.foo.server.connectors.sqs.SqsWriteConnector</serverClassName>
    <sharedClassName>com.foo.client.connectors.sqs.shared.properties.SqsWriteConnectorProperties</sharedClassName>
    <library path="sqsreadwrite-server.jar" type="SERVER"/>
    <library path="sqsreadwrite-shared.jar" type="SHARED"/>
    <library path="sqsreadwrite-client.jar" type="CLIENT"/>

The client class is extending ConnectorSettingsPanel.  I pretty much followed the JavaScript connector as a guide for building this.  Thanks for your help!

**Comment by jonbartels** [comment, marked as answer]
Show your extensions screen please. Is your connector listed in that screen?

Is your connector open-source? Seeing the code on GitHub may help. The active MC plugin devs that I know about (me, Kiran, Kaur, Tony, Chris) lurk on MC Slack. If you want some live help sign on over there!

**Reply by mrjavascript** [reply]
the code is closed for now but will eventually be open source.  I figured out some of the errors by simply restarting Mirth Administrator instead of just logging out.  I'll join the Slack channel, excited to chat about code and pick everyone's brain.
